audioswitch/src/main/java/com/twilio/audioswitch/bluetooth/BluetoothHeadsetManager.kt

```diff
@@ -262,6 +292,23 @@ internal constructor(
                         deviceClass == BluetoothClass.Device.Major.UNCATEGORIZED
             } ?: false
 
+    @SuppressLint("NewApi")
+    private fun hasPermissions(): Boolean {
+        return when (android.os.Build.VERSION.SDK_INT) {
+            in android.os.Build.VERSION_CODES.BASE..android.os.Build.VERSION_CODES.LOLLIPOP_MR1 -> {
+                // before android 23/M checkSelfPermission(..) didn't exist
+                true
+            }
+            in android.os.Build.VERSION_CODES.M..android.os.Build.VERSION_CODES.R -> {
+                PERMISSION_GRANTED == context.checkSelfPermission(Manifest.permission.BLUETOOTH)
+            }
```

Comment:
Would [checkCallingPermission](https://developer.android.com/reference/android/content/Context#checkCallingPermission(java.lang.String)) work here instead?

Reply:
Not according to the Android docs.. IT appears to be used for checking the permissions of an IPC

> One important difference is that if you are not currently processing an IPC, this function will always fail. This is done to protect against accidentally leaking permissions;

Reply:
this might work instead...
                PERMISSION_GRANTED == context.packageManager.checkPermission(
                    Manifest.permission.BLUETOOTH, context.packageName)

Reply:
Okay how about something like this on anything less than Android M?

```
context.checkPermission(permission, android.os.Process.myPid(), Process.myUid())
```

I just want to avoid a hardcoded `true` value.